lib/net/ldap/entry.rb

```diff
@@ -196,4 +196,8 @@ def setter?(sym)
     sym.to_s[-1] == ?=
   end
   private :setter?
+
+  def ==(other)
+    return self['dn'] == other['dn']
```

Comment:
This is a great start but I think we also need to ensure that all the attributes are the same.